src/app.js

```diff
@@ -381,61 +550,220 @@ function initializeCharts() {
+        ...Object.keys(operationData.sharedTree).map(Number)
+    ]);
+    const labels = Array.from(allCellCounts).sort((a, b) => a - b);
+
+    // Calculate averages for SharedMatrix
+    const matrixTimeAverages = labels.map((cells) => {
+        const data = operationData.sharedMatrix[cells];
+        if (data && data.times.length > 0) {
+            return data.times.reduce((sum, time) => sum + time, 0) / data.times.length;
+        }
+        return null;
+    });
+
+    const matrixMemoryAverages = labels.map((cells) => {
+        const data = operationData.sharedMatrix[cells];
+        if (data && data.memory.length > 0) {
+            return data.memory.reduce((sum, mem) => sum + mem, 0) / data.memory.length;
+        }
+        return null;
+    });
+
+    // Calculate averages for SharedTree
+    const treeTimeAverages = labels.map((cells) => {
+        const data = operationData.sharedTree[cells];
+        if (data && data.times.length > 0) {
+            return data.times.reduce((sum, time) => sum + time, 0) / data.times.length;
+        }
+        return null;
     });
-    const memoryAverages = labels.map((cells) => {
-        const memory = dataStore[cells].memory;
-        return memory.length > 0
-            ? memory.reduce((sum, mem) => sum + mem, 0) / memory.length
-            : 0; // Default to 0 if the memory array is empty
+
+    const treeMemoryAverages = labels.map((cells) => {
+        const data = operationData.sharedTree[cells];
+        if (data && data.memory.length > 0) {
+            return data.memory.reduce((sum, mem) => sum + mem, 0) / data.memory.length;
+        }
+        return null;
     });
 
     console.log(`Labels: ${labels}`);
-    console.log(`Time averages: ${averages}`);
-    console.log(`Memory averages: ${memoryAverages}`);
+    console.log(`SharedMatrix time averages: ${matrixTimeAverages}`);
+    console.log(`SharedMatrix memory averages: ${matrixMemoryAverages}`);
+    console.log(`SharedTree time averages: ${treeTimeAverages}`);
+    console.log(`SharedTree memory averages: ${treeMemoryAverages}`);
+
+    // Update the time chart
+    timeChart.data.labels = labels;
+    timeChart.data.datasets[0].data = matrixTimeAverages; // SharedMatrix time
+    timeChart.data.datasets[1].data = treeTimeAverages; // SharedTree time
+    timeChart.update("none");
+
+    // Update the memory chart
+    memoryChart.data.labels = labels;
+    memoryChart.data.datasets[0].data = matrixMemoryAverages; // SharedMatrix memory
+    memoryChart.data.datasets[1].data = treeMemoryAverages; // SharedTree memory
+    memoryChart.update("none");
+}
+
+// Function to remove the SharedMatrix
+async function removeSharedMatrix(sharedMatrix) {
+    // Capture the number of cells before removal
+    const cellCount = sharedMatrix.rowCount * sharedMatrix.colCount;
+    
+    await gc();
+    // Measure memory before the operation
+    const memoryBefore = performance.memory ? performance.memory.usedJSHeapSize : null;
```

Comment:
Nit: you could probably create a helper function like "measureHeapSize" that does these 2 operations in sequence - I think we _always_ want to gc before we take a measurement.

Reply:
The helper function just performs garbage collection, creates two variables, and returns them. Is that too verbose?

Reply:
kk. adding a helper function here.